taroscript/send_test.go

```diff
@@ -301,201 +312,129 @@ func createGenesisProof(t *testing.T, state *spendData) {
 	state.asset2GenesisProof = asset2GenesisProof
 }
 
-func createSpend(t *testing.T, state *spendData, spend taroscript.SpendDelta,
-	full bool) (psbt.Packet, [32]byte, taroscript.SpendDelta,
-	taroscript.SpendCommitments) {
-
-	var (
-		spendAddress     address.Taro
-		receiverStateKey [32]byte
-	)
-
-	if full {
-		spendAddress = state.address2
-		receiverStateKey = state.address2StateKey
-	} else {
-		spendAddress = state.address1
-		receiverStateKey = state.address1StateKey
+func createPacket(addr address.Taro, prevInput asset.PrevID,
+	state spendData, inputSet commitment.InputSet,
+	fullValueInteractive bool) *taropsbt.VPacket {
+
+	inputAsset := inputSet[prevInput]
+	inputs := []*taropsbt.VInput{{
+		PrevID: prevInput,
+	}}
+	outputs := []*taropsbt.VOutput{{
+		Amount: inputAsset.Amount - addr.Amount,
+		ScriptKey: asset.NewScriptKey(
+			&state.spenderScriptKey,
+		),
+		AnchorOutputIndex:       0,
+		IsChange:                true,
+		AnchorOutputInternalKey: &state.spenderPubKey,
+	}, {
+		Amount:                  addr.Amount,
+		ScriptKey:               asset.NewScriptKey(&addr.ScriptKey),
+		AnchorOutputIndex:       receiverExternalIdx,
+		AnchorOutputInternalKey: &addr.InternalKey,
+	}}
+
+	if fullValueInteractive {
+		outputs = []*taropsbt.VOutput{{
+			Interactive: true,
+			Amount:      addr.Amount,
+			ScriptKey: asset.NewScriptKey(
+				&addr.ScriptKey,
+			),
+			AnchorOutputIndex:       receiverExternalIdx,
+			AnchorOutputInternalKey: &state.spenderPubKey,
```

Comment:
Shouldn't this key be `addr.InternalKey` / receiver-controlled in the interactive case?